pkg/gce-pd-csi-driver/node.go

```diff
@@ -311,8 +312,25 @@ func (ns *GCENodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStage
 		return &csi.NodeStageVolumeResponse{}, nil
 	}
 
+	readonly, _ := getReadOnlyFromCapability(volumeCapability)
+	if readonly {
+		options = append(options, "ro")
+		klog.V(4).Infof("CSI volume is read-only, mounting with extra option ro")
+	}
+
 	err = formatAndMount(devicePath, stagingTargetPath, fstype, options, ns.Mounter)
 	if err != nil {
+		// ext4 and ext3 replays the journal which needs write permission. So noload option stops that
```

Comment:
Let's be a bit more precise in the comment. Isn't it more precise to say that if the volume was created by cloning or from a snapshot, the journal might not be in a consistent state even if the fs is otherwise fine. ext3/4 will try to fix that by reloading the journal, which is not possible if the fs is read-only, and which is skipped by noload. We should also mention that this may allow mounting of a corrupted filesystem, but because the mount is read-only no (further) harm will happen.

Reply:
totally agreed